app/routers/google_connect.py

```diff
@@ -0,0 +1,199 @@
+from datetime import datetime
+from fastapi import Depends, APIRouter
+from starlette.responses import RedirectResponse
+
+from google.auth.transport.requests import Request as google_request
+from google.oauth2.credentials import Credentials
+from google_auth_oauthlib.flow import InstalledAppFlow
+from googleapiclient.discovery import build
+
+from app.database.models import Event, User, OAuthCredentials, UserEvent
+from app.database.database import get_db, SessionLocal
+from app.routers.profile import router as profile_router
+from app.config import CLIENT_SECRET_FILE
+from app.routers.event import create_event
+
+SCOPES = ['https://www.googleapis.com/auth/calendar']
+
+router = APIRouter(
+    prefix="/google",
+    tags=["sync"],
+    responses={404: {"description": "Not found"}},
+)
+
+
+@router.get("/sync")
+async def google_sync(session=Depends(get_db)) -> RedirectResponse:
```

Comment:
This function does more than one thing. Please split it into multiple functions

Reply:
This is the main function of the route, how do you want me to split it more?
I don't see whats wrong.

Reply:
???